[Doc] add a section about models naming (#2485)
apollographql/apollo-android: docs/source/essentials/using-aliases.mdx

Since the class hierarchy is flat, having two fields with the same name in two different parts of the query tree could lead to name clashes. Having relay-style pagination with lots of `nodes` or `edges` fields, might typically trigger name clashes. To avoid them, Apollo Android will append a numeric suffix to your models:

```graphql:title=GetContact.graphql
query GetContacts {
  contact(name: "Byron") {
    # The generated model will be named `HomeAddress`
    homeAddress {
      street
    }
  }
  emergencyContact {
    # The generated model will be named `HomeAddress1` by default
    homeAddress {
      street
    }
  }
}
```

Suffixing avoids the name clash but feels a bit out of place and it's hard to remember what `HomeAddress1` is about. To workaround this issue, you can alias the field:

```graphql:title=GetContact.graphql
query GetContacts {
  contact(name: "Byron") {
    # The generated model will be named `HomeAddress`
    homeAddress {
      street
    }
  }
  emergencyContact {
    # With an alias, the generated model will be named `EmergencyAddress`
    emergencyAddress: homeAddress {
      street
    }
  }
}
```

## Using aliases to fix singularization rules

> The [singularization code](https://github.com/martinbonnin/apollo-android/blob/master/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Inflector.kt) works most of time but does not contain all the exceptions in the english dictionnary.

Reviewer: ```suggestion
The [singularization code](https://github.com/martinbonnin/apollo-android/blob/master/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Inflector.kt) works most of time but does not contain all the exceptions in the English dictionnary.
```

> If you encouter one, please let me know so I can put a valid example here (we usually fix those issues as we go but there certainly are a few left in the wild). Until we fix it in the singularization code, you can use an alias to control the name of the model in your code.

Reviewer: The second half of the sentence uses we/us. The first half can do that too.

```suggestion
If you encouter one, please let us know so we can put a valid example here (we usually fix those issues as we go but there certainly are a few left in the wild). Until we fix it in the singularization code, you can use an alias to control the name of the model in your code.
```

Author: Good catch, thanks!
